src/mca.jl

```diff
@@ -0,0 +1,354 @@
+# Correspondence Analysis and Multiple Correspondence Analysis
+
+# Needed for the plotting function below
+#using Printf, PyPlot
+
+#==
+References:
+https://personal.utdallas.edu/~herve/Abdi-MCA2007-pretty.pdf
+https://www.stata.com/manuals/mvmca.pdf
+https://www.stata.com/manuals/mvca.pdf
+https://en.wikipedia.org/wiki/Multiple_correspondence_analysis
+https://pca4ds.github.io
+https://maths.cnam.fr/IMG/pdf/ClassMCA_cle825cfc.pdf
+==#
+
+"""
+Correspondence Analysis
+"""
+mutable struct CA{T<:Real} <: LinearDimensionalityReduction
+
+    # The data matrix
+    Z::Array{T}
+
+    # The residuals
+    R::Array{T}
+
+    # Row and column masses (means)
+    rm::Vector{T}
+    cm::Vector{T}
+
+    # The standardized residuals
+    SR::Array{T}
+
+    # Object scores
+    F::Array{T}
+
+    # Variable scores
+    G::Array{T}
+
+    # Inertia (eigenvalues of the indicator matrix)
+    I::Vector{T}
+end
+
+# Constructor
+
+function CA(X)
+
+    # Convert to proportions
+    X = X ./ sum(X)
+
+    # Calculate row and column means
+    r = sum(X, dims = 2)[:]
+    c = sum(X, dims = 1)[:]
+
+    # Center the data matrix to create residuals
+    R = X - r * c'
+
+    # Standardize the data matrix to create standardized residuals
+    Wr = Diagonal(sqrt.(r))
+    Wc = Diagonal(sqrt.(c))
+    SR = Wr \ R / Wc
+
+    T = eltype(X)
+    return CA(X, R, r, c, SR, zeros(T, 0, 0), zeros(T, 0, 0), zeros(T, 0))
+end
+
+function fit!(ca::CA, d::Int)
+
+    # Get the object factor scores (F) and variable factor scores (G).
+    P, D, Q = svd(ca.SR)
+    Dq = Diagonal(D)[:, 1:d]
+
+    # Check that there are no repeated non-zero eigenvalues.
+    d = diff(D[D .> 1e-10])
+    if maximum(d) >= -1e-10
+        @warn("The indicator matrix has repeated non-zero eigenvalues")
+    end
+
+    Wr = Diagonal(sqrt.(ca.rm))
+    Wc = Diagonal(sqrt.(ca.cm))
+    ca.F = Wr \ P * Dq
+    ca.G = Wc \ Q * Dq
+
+    # Get the eigenvalues
+    ca.I = D .^ 2
+end
+
+function fit(::Type{CA}; X::AbstractMatrix, d::Int = 5)
```

Comment:
`X` should be a second positional argument. Use `maxoutdim` keyword parameter to specify maximal output dimension.